### PR TITLE
Refactoring the filter in CM

### PR DESCRIPTION
**Pull request by mfrachet**
### What does it do?

It modifies the filter in the CM to show something that looks like the media library filters

### Why is it needed?

Consistency

### How to test it?

Go to the CM and use the filter. You should see a dropdown down with two selects and an input. When validating, the URL should change with the good filter

### CODE REVIEW

File: packages/core/content-manager/admin/src/components/AttributeFilter/utils/getAttributeType.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import get from 'lodash/get';
+
+const getAttributeType = (attributeName, contentType, metaData) => {
```
Comment (soupette) [inline packages/core/content-manager/admin/src/components/AttributeFilter/utils/getAttributeType.js:3]: The string input does not exist I prefer to return a text instead WDYT?

**Review by soupette — CHANGES_REQUESTED** [review]
Also can you make you PR against the `chore/move-cm-in-core` branch instead?

File: packages/core/content-manager/admin/src/components/AttributeFilter/GenericInput.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      return <InputText onChange={e => onValueChange(e.target.value)} value={value} {...rest} />;
+  }
+}
+
+GenericInput.defaultProps = {
+  value: undefined,
+};
+
+GenericInput.propTypes = {
+  onValueChange: PropTypes.func.isRequired,
```
[the remark this answers is not in the document]
Comment (mfrachet) [reply packages/core/content-manager/admin/src/components/AttributeFilter/GenericInput.js:60]: I will put onChange. onFilterChange is a parent responsibility, not a GenericInput one to me 😊

File: packages/core/content-manager/admin/src/components/AttributeFilter/utils/getAttributeType.js (hunk 1 above)
Comment (mfrachet) [reply packages/core/content-manager/admin/src/components/AttributeFilter/utils/getAttributeType.js:3]: I did not understand the comment, sorry 😅

**Review by HichamELBSI — APPROVED** [review]
LGTM

File: packages/core/admin/admin/src/content-manager/components/AttributeFilter/GenericInput.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      return <Select onChange={e => onChange(e.target.value)} value={value} {...rest} />;
+
+    case 'date':
+    case 'timestamp':
+    case 'timestampUpdate': {
+      const momentValue = moment(value);
+
+      return (
+        <DateWrapper type={type}>
+          <DatePicker onChange={e => onChange(e.target.value._d)} value={momentValue} {...rest} />
```
Comment (soupette) [inline packages/core/admin/admin/src/content-manager/components/AttributeFilter/GenericInput.js:20]: Here we need to remove the remove the timezone:


```
onChange={e => {
 if (e.target.value) {
  onChange(e.target.value.format('YYYY-MM-DD'));
 }
}
```

File: packages/core/admin/admin/src/content-manager/components/AttributeFilter/GenericInput.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        </DateWrapper>
+      );
+    }
+
+    case 'datetime': {
+      const momentValue = moment(value);
+
+      return (
+        <DateWrapper type={type}>
+          <DateTime onChange={e => onChange(e.target.value)} value={momentValue} {...rest} />
```
Comment (soupette) [inline packages/core/admin/admin/src/content-manager/components/AttributeFilter/GenericInput.js:30]: In the same way:

```
onChange={e => {
  if (e.target.value) {
    onChange(e.target.value.toISOString());
  }
}
```

**Review by soupette — CHANGES_REQUESTED** [review]
Awesome work! 
Well done

File: packages/core/admin/admin/src/content-manager/components/AttributeFilter/GenericInput.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+    case 'date':
+    case 'timestamp':
+    case 'timestampUpdate': {
+      const momentValue = moment(value);
+
+      return (
+        <DateWrapper type={type}>
+          <DatePicker
+            onChange={e => onChange(e.target.value.format('YYYY-MM-DD'))}
```
Comment (soupette) [inline packages/core/admin/admin/src/content-manager/components/AttributeFilter/GenericInput.js:21]: You need to make a condition here as the input can be cleared.

```
if (e.target.value) {
  onChange(....)
}
```

**Review by soupette — CHANGES_REQUESTED** [review]

**Review by soupette — APPROVED** [review]
LGTM

**Pull request merged**